src/OrchardCore.Modules/OrchardCore.OpenId/Validators/OpenIdUrlValidator.cs

```diff
@@ -1,20 +1,24 @@
 using System;
 using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
 
 namespace OrchardCore.OpenId.Validators
 {
     public static class OpenIdUrlValidator
     {
-        public static IEnumerable<ValidationResult> ValidateUrls(string memberName, string member)
+        public static IEnumerable<ValidationResult> ValidateUrls<T>(this ValidationContext context, string memberName, string member)
         {
             if (member != null)
             {
+                IStringLocalizer<T> s = null;
                 foreach (var url in member.Split(new[] { ' ', ',' }, StringSplitOptions.RemoveEmptyEntries))
                 {
-                    if (!Uri.TryCreate(url, UriKind.Absolute, out Uri uri) || !uri.IsWellFormedOriginalString())
+                    if (!Uri.TryCreate(url, UriKind.Absolute, out var uri) || !uri.IsWellFormedOriginalString())
                     {
-                        yield return new ValidationResult($"{url} is not wellformed", new[] { memberName });
+                        s ??= context.GetService<IStringLocalizer<T>>();
```

Comment:
Nothing prevent to move the declaration above

```csharp
var s ??= context.GetService<IStringLocalizer<T>>();
```

Reply:
you cannot do that.